map_nav_node:
  ros__parameters:
    wms:
      url: 'http://localhost:8080/wms'
      version: '1.1.1'
      layer: 'Imagery'
      srs: 'EPSG:4326'  # don't change this setting, internal logic may often implicitly assume EPSG:4326
      request_timeout: 5
    misc:
      mock_gps_selection: 1 # declared by MockGPSNode, not by base class
      export_position: 'position.json' # declared by MockGPSNode, not by base class
      export_projection: 'projection.json' # declared by MockGPSNode, not by base class
      max_pitch: 10  # 30  # Used by _should_match
      variance_estimation_length: 20
      min_match_altitude: 80
      blur_threshold: 1.5  # standard deviations
      blur_window_length: 100
      visual_odometry: True
      visual_odometry_update_t_threshold: 0.3  # multiple of focal length, translation threshold
      visual_odometry_update_r_threshold: 0.3  # radians, rotation threshold
    map_update:
      initial_guess: # Values from ksql_airport.world file
        - 37.523640
        - -122.255122
      update_delay: 1  # Keep low (e.g. 1 second) if vehicle flying fast or gimbal_projection: True
      default_altitude: 130
      gimbal_projection: True
      max_pitch: 30  # Applies if gimbal_projection: True, used by _should_update_map
      max_map_radius: 1000
      map_radius_meters_default: 400
      update_map_area_threshold: 0.7
    map_matcher:
      #params_file: 'config/superglue_params.yml'  # For parsing args for matcher's static initializer method
      params_file: 'config/loftr_params.yml'
    vo_matcher:
      params_file: 'config/orb_params.yml'
